src/plan/plan_constraints.rs

```diff
@@ -13,8 +13,6 @@ pub struct PlanConstraints {
     pub gc_header_words: usize,
     pub num_specialized_scans: usize,
     pub max_non_los_copy_bytes: usize,
-    pub needs_log_bit_in_header: bool,
```

Comment:
I thought these two constraints are not meaningful in Rust MMTk.
The decision on whether the log-bit is in-header or on-side is made by the VM binding.

Unless it only means `needs_log_bit`, rather than `needs_log_bit_in_header`.
In that case, we can remove the bit number and keep `needs_log_bit` without in_header.
Still, as this is not used anywhere, I'm not sure whether it's a good idea to leave it there

Reply:
If it is `needs_log_bit`, is it possible to implement it with the new metadata implementation? It would be an issue if we can't implement it with the refactored code. 

> It is not used anywhere

For the code we have ported from Java MMTk, it is not used. For the code we haven't ported, those may be useful.

Reply:
Yes, we can implement it on the same way as we do for other per-object metadata.

If it is used by the code we haven't ported yet, it still is unused. 
I think such code pieces make the code harder to read and understand and there is no reason to include them in the code-base, until we decide to use them.

Reply:
You still have an unlog bit spec in the ObjectModel. But you removed any code that may set the bit. What will happen is that when someone implemets a plan that actually need this bit, he would use the spec but find it is never set, and after frustrating debugging, he would find that we have the spec and we have the policy, but the policy does not set the bit due to this PR.

Reply:
Right, I should remove the unlogged-bit spec from the ObjectModel as well